src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -140,46 +140,30 @@ private DirectoryInfo GetTestDestinationDirectory(string callingMethod, string i
 
         private IEnumerable<FileInfo> LoadInventory(FileInfo file)
         {
+            file.Refresh();
             if (!file.Exists)
             {
-                return Enumerable.Empty<FileInfo>();
+                throw new InvalidOperationException("Inventory file should exist.");
             }
 
             var inventory = new List<FileInfo>();
-
-            var lines = file.OpenText();
-
-            while (lines.Peek() > 0)
+            foreach (var p in File.ReadAllLines(file.FullName))
             {
-                inventory.Add(new FileInfo(lines.ReadLine()));
+                inventory.Add(new FileInfo(p));
             }
 
             return inventory;
         }
 
         private void SaveInventory(FileInfo file, IEnumerable<FileInfo> inventory)
         {
-            FileUtility.ReplaceWithLock(
-                filePath =>
-                {
-                    if (!_dataDirectory.Exists)
-                    {
-                        _dataDirectory.Create();
-                    }
+            _dataDirectory.Refresh();
```

Comment:
The intent of the caller is to get some inventory and not have to care about locking, so a method that returns the inventory would be a better factoring than capturing it in a closure that also handles locking.

As to how to do the locking, the side-effecting code shouldn't be callable without a lock. Factoring out a class that can't be constructed without a lock instance, and moving `SaveInventory` to that class, would enforce this in a way that naming conventions can't.